gax/tests/errors.rs

```diff
@@ -66,3 +69,49 @@ fn downcast() -> Result<(), Box<dyn std::error::Error>> {
     assert!(inner_err.is_none());
     Ok(())
 }
+
+#[tokio::test]
+async fn client_http_error() -> Result<(), Box<dyn std::error::Error>> {
+    let http_resp = http::Response::builder()
+        .header("Content-Type", "application/json")
+        .status(400)
+        .body(r#"{"error": "bad request"}"#)?;
+
+    // Into reqwest response, like our clients use.
+    let resp: reqwest::Response = http_resp.into();
+
+    assert!(resp.status().is_client_error());
+
+    let status = resp.status().as_u16();
+    let headers = gax::error::convert_headers(resp.headers());
+    let body = resp.bytes().await?;
+
+    let http_err = HttpError::new(status, headers, Some(body));
+    assert!(http_err.status_code() == 400);
+    assert!(http_err.headers()["content-type"] == "application/json");
+    assert!(http_err.payload().unwrap() == r#"{"error": "bad request"}"#.as_bytes());
+    Ok(())
+}
+
+#[test]
+fn http_error_to_status() -> Result<(), Box<dyn std::error::Error>> {
+    let json = serde_json::json!({
+        "code": 9,
+        "message": "msg",
+        "details": [
+            {"violations": [{"type": "type", "subject": "subject", "description": "desc"}]},
+        ]
+    });
+    let http_err = HttpError::new(
+        400,
+        HashMap::from_iter([("content-type".to_string(), "application/json".to_string())]),
+        Some(json.to_string().into()),
+    );
+
+    let status: Status = http_err.try_into()?;
```

Comment:
Should we have a test for `try_into()` failures?  More specifically: do we want to verify it fails to convert something like:

```html
<!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 404 (Not Found)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>404.</b> <ins>That’s an error.</ins>
  <p>The requested URL <code>/v7/projects/coryan-test/instances/test001/tables/test:readRows</code> was not found on this server.  <ins>That’s all we know.</ins>
  ```

Reply:
added a failure case.